packages/api/src/models/userFarmModel.js

```diff
@@ -140,6 +140,39 @@ class userFarm extends Model {
       .first();
   }
 
+  /**
+   * Checks if the user exists on a particular farm.
+   * @param user_id
+   * @param farm_id
+   * @return {Objection.QueryBuilder<userFarm, userFarm>}
+   * @static
+   * @async
+   */
+  static async checkIfUserExistsOnFarm(user_id, farm_id) {
+    return userFarm.query().where({ user_id, farm_id }).first();
+  }
+
+  /**
+   * Gets a userFarm record by email.
+   * @param email
+   * @param farm_id
+   * @param trx - optional transaction
+   * @return {Objection.QueryBuilder<userFarm, userFarm>}
+   */
+  static async getUserFarmByEmail(email, farm_id, trx) {
+    const transaction = trx ?? (await this.startTransaction());
+    const result = await userFarm
+      .query(transaction)
+      .join('users', 'userFarm.user_id', '=', 'users.user_id')
+      .join('farm', 'farm.farm_id', '=', 'userFarm.farm_id')
+      .join('role', 'userFarm.role_id', '=', 'role.role_id')
+      .where({ 'users.email': email, 'userFarm.farm_id': farm_id })
+      .first()
+      .select('*');
+    if (!trx) {
+      await transaction.commit();
+    }
+    return result;
```

Comment:
Seems to be a syntax error here.

Reply:
Thanks for catching that! I've fixed this now